src/hotspot/share/gc/shared/referenceProcessor.hpp

```diff
@@ -120,14 +119,8 @@ class DiscoveredListIterator {
   // Remove the current reference from the list
   void remove();
 
-  // Make the referent alive.
-  inline void make_referent_alive() {
-    if (UseCompressedOops) {
-      _keep_alive->do_oop((narrowOop*)_referent_addr);
-    } else {
-      _keep_alive->do_oop((oop*)_referent_addr);
-    }
-  }
+  // Apply the keep_alive function to the referent address.
+  void make_referent_alive();
```

Comment:
I wonder if moving this from the `.hpp` to `.cpp` has performance implications for callers. Maybe move to `.inline.hpp`?

Reply:
The reason for moving it out of the .hpp was of course that the
change to call java_lang_ref_Reference::referent_addr_raw needs to #include
javaClasses.inline.hpp.

I don't see this move having any measureable performance difference, and not
even sure what the sign of any change might be.  A better refactoring might
be to package up the common remove/make_referent_alive/move_to_next sequence.

Reply:
Yeah, it is probably only affects the path that is infrequently taken, i.e. marking through the resurrected referents during weak reference processing, that is probably only finalizers. It is fine to have it in `.cpp`.